plugins/jobs/src/js/types/Job.ts

```diff
@@ -201,24 +199,30 @@ export const JobFieldResolvers = {
 };
 
 function AddStatusToHistoryJobRuns(
-  job: MetronomeJobDetailResponse
+  job: MetronomeGenericJobResponse
 ): JobRunConnection {
-  const { successfulFinishedRuns, failedFinishedRuns } = job.history;
+  let successfulFinishedRunsWithStatus: JobHistoryRun[] = [];
+  let failedFinishedRunsWithStatus: JobHistoryRun[] = [];
+  const activeRuns = job.activeRuns || [];
 
-  const successfulFinishedRunsWithStatus: JobHistoryRun[] = successfulFinishedRuns.map(
-    run => ({ ...run, status: "COMPLETED" as MetronomeJobStatus }) // TODO: investiagte why we need to cast this
-  );
+  if (isMetronomeJobDetailResponse(job)) {
+    const { successfulFinishedRuns, failedFinishedRuns } = job.history;
+
+    successfulFinishedRunsWithStatus = successfulFinishedRuns.map(
+      run => ({ ...run, status: "COMPLETED" as MetronomeJobStatus }) // TODO: investiagte why we need to cast this
+    );
 
-  const failedFinishedRunsWithStatus: JobHistoryRun[] = failedFinishedRuns.map(
-    run => ({
+    failedFinishedRunsWithStatus = failedFinishedRuns.map(run => ({
       ...run,
       status: "FAILED" as MetronomeJobStatus
-    })
-  );
+    }));
+  }
 
-  return JobRunConnectionTypeResolver([
-    ...job.activeRuns,
-    ...successfulFinishedRunsWithStatus,
-    ...failedFinishedRunsWithStatus
-  ]);
+  let retval: Array<MetronomeActiveJobRun | JobHistoryRun> = [];
+  retval = retval.concat(
```

Comment:
I would like us not to go with retval here

Reply:
I agree, but because of the different types I wasnt able to do it inline 😩

Reply:
Well, that's because there's a type mismatch :)

`JobRunConnectionTypeResolver` expects `MetronomeActiveJobRun[] | JobHistoryRun[]` and you are trying to pass `Array<MetronomeActiveJobRun | JobHistoryRun>`. So I'd say we should rather fix the type the `JobRunConnectionTypeResolver` expects instead of hacking with `retval`

Reply:
uh, you are right, I'll fix that 👍

Reply:
Hmmm… https://github.com/dcos/dcos-ui/blob/b6419cce6f4ad95ec710d8faf2680ef0d2e00b09/plugins/jobs/src/js/types/JobRunConnection.ts#L25 @nLight @DanielMSchmidt 🤔

Reply:
all three are arrays, so you end up with an array of arrays, where's the function expects an array of items.
```js
  return JobRunConnectionTypeResolver([
    ...activeRuns,
    ...successfulFinishedRunsWithStatus,
    ...failedFinishedRunsWithStatus
  ]);
```

Reply:
This code actually doesn't complain 🤷‍♂️